### DISCUSSION TITLE
MPICH and Slurm job allocation

### DISCUSSION THREAD

**Discussion by rayaprolu143** [idaholab/moose#19988] [Q&A Getting Started] [answered]
Hi,

I want to first thank the MOOSE team for making the installation so easy to perform through mamba. However, I am facing the issue of not being able to run the jobs with the Slurm workload manager. I receive an error that is as follows: "Error: node list format not recognized. Try using '-hosts=<hostnames>' ". The culprit is the 3.3.2 version of MPICH used in the environment packaged in moose-tools. To better understand the issue at hand, I ask the reader to follow this link  https://lists.mpich.org/pipermail/discuss/2019-December/005835.html . I request the MOOSE community to provide answers to the following questions:

1. Is there a way to circumvent the issue with the existing MPICH 3.3.2 version?
2. Is it possible to upgrade MPICH in the package to a later version?

Thanks,
Sreekar

**Comment by GiudGiud** [comment, marked as answer]
Hello

The workflow for installing moose on computing clusters does not use mamba, it's detailed here: https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html
The idea is that a custom install will make use of optimized compilers on your cluster, which the mamba install cannot do.

1. Did you try the solution listed in that thread you posted? https://lists.mpich.org/pipermail/discuss/2019-December/005839.html
2. @milljm will know

Guillaume

**Reply by rayaprolu143** [reply]
Hi,

I installed the MOOSE following the HPC cluster instructions.

Thank you

**Reply by milljm** [reply]
Glad you got it sorted, and sorry for the delayed response. I only just saw an email come in.

We are in the process of updating MPICH to 3.4.2 in fact: https://github.com/idaholab/moose/pull/19456. But as GuidGuid stated, we try and refrain from asking users to install our Conda packages on High Performance Computing clusters. The compilers which come with the Conda (Mamba) install are not optimized. Meaning they are designed to run on just about any modern CPU architecture. e.g:

```
# equal to or greater than micro-architecture
-mtune=haswell
# Intel/AMD CPUs dating back to 2013
```
